src/manifest-parser.js

```diff
@@ -0,0 +1,157 @@
+import { parse as parseMpd } from 'mpd-parser';
+import { Parser as M3u8Parser } from 'm3u8-parser';
+import { simpleTypeFromSourceType } from '@videojs/vhs-utils/dist/media-types';
+import resolveUrl from '@videojs/vhs-utils/dist/resolve-url.js';
+
+/**
+ * Parses an m3u8 string using m3u8-parser.
+ *
+ * @param {string} manifestString
+ *        M3U8 manifest string
+ *
+ * @return {Object}
+ *         Manifest object
+ */
+export const parseM3u8 = (manifestString) => {
+  const parser = new M3u8Parser();
+
+  parser.push(manifestString);
+  parser.end();
+
+  return parser.manifest;
+};
+
+/**
+ * Adds a resolvedUri property for each URI in the segment.
+ *
+ * @param {Object} segment
+ *        The segment
+ * @param {string} baseUri
+ *        URI of the playlist the segment is from
+ */
+export const resolveSegmentUris = (segment, baseUri) => {
+  if (!segment.resolvedUri) {
+    segment.resolvedUri = resolveUrl(baseUri, segment.uri);
+  }
+  if (segment.key && !segment.key.resolvedUri) {
+    segment.key.resolvedUri = resolveUrl(baseUri, segment.key.uri);
+  }
+  if (segment.map && !segment.map.resolvedUri) {
+    segment.map.resolvedUri = resolveUrl(baseUri, segment.map.uri);
+  }
+};
+
+/**
+ * Adds properties to a master manifest object that are needed by videojs-concat. For
+ * instance, resolvedUri for playlists, since videojs-concat may need to request the
+ * playlists.
+ *
+ * Note that videojs/http-streaming may add more properties to the manifest object
+ * afterwards, but if they're not needed by videojs-concat, videojs-concat leaves that
+ * job to videojs/http-streaming.
+ *
+ * @param {Object} master
+ *        Master manifest object
+ */
+export const addPropertiesToMaster = (master) => {
+  for (let i = 0; i < master.playlists.length; i++) {
+    const playlist = master.playlists[i];
+
+    if (!playlist.uri) {
```

Comment:
Maybe we should just do this in mpd-parser now, we have a similar comment in vhs in the function with the same name.